lib/run_loop/simctl.rb

```diff
@@ -331,6 +374,17 @@ def uninstall(device, app, timeout)
 
 ]
       end
+
+      app_container = app_container(device, app.bundle_identifier)
+      if app_container
+        RunLoop.log_debug("After install, app container exists")
+        RunLoop.log_debug("Rebooting the device")
+        reboot(device)
+        if app_container(device, app.bundle_identifier)
+          raise "simctl uninstall succeeded, but simctl says app is still installed"
```

Comment:
Maybe use a `PartialAppInstallationError` as I suggested above?

Reply:
This is not always the same as the "cached app problem".

On the file system, the .app bundle usually gone, but the `simctl` command still reports the app as having an app and data container.  The FBSimulatorControl private APIs report the same information about the existence of the app and data containers.

A "reboot" of the device appears to reset (FrontBoard?) whatever mechanism is caching the container information.

Reply:
uh, trixy. Maybe put that in a code comment?

Reply:
Added verbose logging to distinguish between "cached app problem" and "simctl needs reboot problem"